src/node_dotenv.cc

```diff
@@ -48,14 +48,19 @@ void Dotenv::SetEnvironment(node::Environment* env) {
   for (const auto& entry : store_) {
     auto key = entry.first;
     auto value = entry.second;
-    env->env_vars()->Set(
-        isolate,
-        v8::String::NewFromUtf8(
-            isolate, key.data(), NewStringType::kNormal, key.size())
-            .ToLocalChecked(),
-        v8::String::NewFromUtf8(
-            isolate, value.data(), NewStringType::kNormal, value.size())
-            .ToLocalChecked());
+
+    auto existing = env->env_vars()->Get(key.data());
+
+    if (existing.IsNothing()) {
```

Comment:
Can you add an exception for `NODE_OPTIONS` and add a test for it?

Reply:
FYI, do we have a mechanism for stopping infinite loop that can be caused from this:
```
# .env
NODE_OPTIONS='--env-file=./.env'
```

```bash
$ NODE_OPTIONS='--env-file=./.env' node something.js 
```

Reply:
There won't be any infinite loop. Ref: https://github.com/nodejs/node/commit/769823e57c90fc15e1854ae0d91a1b9a8d2a0c52#diff-6c1386cbc6e4a11941649af31a0257d5c3e40b5f41223cdddf87ab2e984b705aR856

Reply:
Done, let me know if that looks ok!

Reply:
@philnash should we add a test that overriding `NODE_OPTIONS` works?

Reply:
@atlowChemi I actually did implement this and then, after [this investigation](https://github.com/nodejs/node/pull/49424#issuecomment-1704511079), backed that change out again. My reasoning is all in that comment.